File: eknjiga_mobile_app/lib/models/commentReaction.dart
class CommentReaction {
  final int userId;
  final int? commentId;
  final int? commentAnswerId;
  final bool isLike;

  CommentReaction({
    required this.userId,
    this.commentId,
    this.commentAnswerId,
    required this.isLike,
  });

  Map<String, dynamic> toJson() => {
    'userId': userId,
    'commentId': commentId,
    'commentAnswerId': commentAnswerId,
    'isLike': isLike,
  };
}

File: eKnjiga/eKnjiga.Services/Database/CommentReaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eKnjiga.Services.Database
{
    public class CommentReaction
    {
        [Key]
        public int Id { get; set; }

        public int? CommentId { get; set; }
        [ForeignKey("CommentId")]
        public Comment? Comment { get; set; }

        public int? CommentAnswerId { get; set; }
        [ForeignKey("CommentAnswerId")]
        public CommentAnswer? CommentAnswer { get; set; }

        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        // true = like, false = dislike
        public bool IsLike { get; set; }
    }
}
